### ISSUE TITLE
Rllib - is_training never true

### ISSUE THREAD

**Issue by guzmo** [ray-project/ray#35833]
### What happened + What you expected to happen

1. I've been testing out to create a custom model with Keras(tf2) and PPO. When I'm logging the forward method is_training never gets set to true. I've also tried to add update_ops() and custom_loss() but those never gets triggered.
2. I expected is_training to be true during training, and false during evaluation.

### Versions / Dependencies

Ray/RLlib 2.4.0

### Reproduction script

If needed I can send one, but it's basically happening anytime I create a custom model with Keras that has layers that needs to be "trained"

### Issue Severity

High: It blocks me from completing my task.

**Comment by guzmo** [comment]
I've digged around more in the code, and training gets set to true to the SampleBatch. Apparantly this code in the SampleBatch class always goes in to the if statement and gets False?
```
@property
    def is_training(self):
        if self.get_interceptor is not None and isinstance(self._is_training, bool):
            if "_is_training" not in self.intercepted_values:
                self.intercepted_values["_is_training"] = self.get_interceptor(
                    self._is_training
                )
            # Always goes here with false?
            return self.intercepted_values["_is_training"]
        return self._is_training
```

And in ModelV2 __call__() it prints true, but after the if/else is_training is false. So I guess the copy is not working?
```
print(input_dict.is_training)
if isinstance(input_dict, SampleBatch):
      restored = input_dict.copy(shallow=True)
else:
      restored = input_dict.copy()
print(restored.is_training)

```

**Comment by guzmo** [comment]
Added the is_training flag specifically in the copy() method of SampleBatch and now it seems to work as I expected it too.

**Comment by Rohan138** [comment]
Hi @guzmo, does your fix work for you? If not, please let us know.

Also, for working with custom policies and models, I would recommend switching to the new [RLModule](https://docs.ray.io/en/latest/rllib/rllib-rlmodule.html) API instead; we will eventually deprecate the `PolicyV2` and `ModelV2` classes in favor of it. In your use case, you would separately modify the `RLModule.forward_*` methods:
- _forward_train(): Forward pass for training.
- _forward_inference(): Forward pass for inference.
- _forward_exploration(): Forward pass for exploration.

**Comment by guzmo** [comment]
Yeah, it seems to send is_training=True. I also changed to tf1 to make things work.
I would like to test the new RLModule but it seems a bit too early and it lacks a bit of documentation. Had problem understanding the old one and debug it to know if everything worked or not :P but I might give it a try soon.
